wgengine/magicsock/magicsock.go

```diff
@@ -1727,7 +1727,17 @@ func (c *Conn) runDerpReader(ctx context.Context, derpFakeAddr netip.AddrPort, d
 		case derp.HealthMessage:
 			health.SetDERPRegionHealth(regionID, m.Problem)
 		case derp.PeerGoneMessage:
-			c.removeDerpPeerRoute(key.NodePublic(m), regionID, dc)
+			switch m.Reason {
+			case derp.PeerGoneReasonDisconnected:
+				// Do nothing.
+			case derp.PeerGoneReasonNotHere:
+				c.logf("[unexpected] magicsock: derp-%d does not know about peer %s, removing route",
```

Comment:
May want to increment a client metric if this is an unexpected/interesting thing to happen (there's already a few DERP-related metrics in this file).

Reply:
Good idea! I added two new metrics, noticed two arguments were swapped for the log message for the unknown reason case and fixed that, and tested the not here path. If you can double check the logf() after metricRecvDiscoDERPPeerGoneUnknown.Add(1), I would appreciate it.